### PR TITLE
Added fix for issue smart-204, version not being updated in the Add ...

### PR DESCRIPTION
**Pull request by iuliana**
to catalog modal. (More like, refactored a little to the fix can be aplied on implementation provided by `composerOverrides`)

### CODE REVIEW

File: ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js [hunk 1]
```diff
@@ -68,6 +68,30 @@ export function saveToCatalogModalDirective($rootScope, $uibModal, $injector, $f
         }
     }
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js:70]: I would not include so many custom calls from the override.
 
Suggestion:

```
export function saveToCatalogModalDirective($rootScope, $uibModal, $injector, $filter, composerOverrides, blueprintService) {
    let modalDirective = this;

    modalDirective.getConfigNameFromEntity = (entity) => {
    
    // use $scope directly
    // code omitted ...

    };

    modalDirective.getConfigBundleNameFromScope = () => {

    // use $scope directly
    // code omitted ...

    };

    modalDirective.getSuggestedSymbolicNameFromBlueprint = (entity, metadata) => {

    // use $scope directly
    // code omitted ...

    };

    modalDirective.updateCatalogConfig = (config, $element) => {

    // use $scope directly
    // code omitted ...

    }

    // allow downstream to configure directive and scope here
    (composerOverrides.configureCatalog || () => {})(modalDirective, $scope);

    return = {

    // code omitted ...
```

Customize methods defined on `modalDirective` instance in the branding part and call them accessing instance of the directive `modalDirective.getConfigNameFromEntity(entity)`

**Review by algairim — CHANGES_REQUESTED** [review]

**Comment by ahgittin** [comment]
methods starting `get...` i expect to return the value, not write the value in scope.  `populateXxxx` or `setXxxx` might be better choices.

otherwise looks good.

i also prefer @algairim 's pattern but don't feel strongly.

File: ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js (hunk 1 above)
Comment (algairim) [reply ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js:70]: Correction: it is better to define methods above at the controller level, not at directive:
```
export function saveToCatalogModalDirective($rootScope, $uibModal, $injector, $filter, composerOverrides, blueprintService) {
    return {
        restrict: 'E',
        templateUrl: function (tElement, tAttrs) {
            return tAttrs.templateUrl || TEMPLATE_URL;
        },
        scope: {
            config: '=',
        },
        controller: ['$scope','composerOverrides', saveToCatalogModalController],
        link: link
    };

    function saveToCatalogModalController($scope, composerOverrides) {
        let modalController = this;

        modalController.getConfigNameFromEntity = (entity) => {

            // use $scope directly
            // code omitted ...

        };
        
        // etc. ...

        // allow downstream to configure directive and scope here
        (composerOverrides.configureCatalogModal || function () {})(modalController, $scope);
    }
```

File: ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js [hunk 2]
```diff
@@ -93,22 +117,10 @@ export function saveToCatalogModalDirective($rootScope, $uibModal, $injector, $f
...
-                }
+                (composerOverrides.getConfigNameFromEntity || getConfigNameFromEntity)(entity, $scope);
+                (composerOverrides.getSuggestedSymbolicNameFromBlueprint || getSuggestedSymbolicNameFromBlueprint)(entity, metadata, $scope);
                 (composerOverrides.getSuggestedVersionToSaveFromBlueprint || getSuggestedVersionToSaveFromBlueprint)(entity, metadata, $scope);
-                if (!$scope.config.bundle) {
-                    if ($scope.config.symbolicName) {
-                        $scope.config.bundle = $scope.config.symbolicName;
-                    }
-                }
+                (composerOverrides.getConfigBundleNameFromScope || getConfigBundleNameFromScope)($scope);
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js:123]: Grouping them together looks better.

**Review by algairim — APPROVED** [review]

File: ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js (hunk 1 above)
Comment (iuliana) [reply ui-modules/blueprint-composer/app/components/catalog-saver/catalog-saver.directive.js]: Tried it, the pattern cannot be applied to this implementation.

**Comment by iuliana** [comment]
The test failing is one of our flaky ones.

**Pull request merged**